### ISSUE TITLE
Issues with fonts with precompiled package

### ISSUE THREAD

**Issue by akiross** [sciapp/gr#69]
Hello,
I would like to use the C interface to gr framework. I want to work on a confined development environment, so I downloaded a package for redhat from the [downloads](https://gr-framework.org/downloads/) (I'm using Fedora 28) and unpacked it.

I can compile and link a simple application against it, but I am unable to obtain the correct output due to missing fonts:

```
GKS: file open error (/tmp/gr/fonts/gksfont.dat)
open: No such file or directory
GKS: could not open font file: /tmp/gr/fonts/NimbusSanL-Regu.pfb
```

I tried fixing this by linking the font directory to `/tmp/gr/fonts`, which partially fixes the issue as no error messages are printed now, but the output is definitely wrong, as you can see from the picture below

![gr_framework_font_issue](https://user-images.githubusercontent.com/884092/49080873-53f48880-f245-11e8-840b-7f6f55b38d39.png)

How can I fix it? I tried searching for an API call to change font dir or load custom fonts, but I couldn't find it.

Thanks and best regards

**Comment by jheinen** [comment]
Could you try to set `GRDIR` to `/tmp/gr` (where you unpacked it) and use another output driver (than X11):
```
export GRDIR=/tmp/gr
export GKS_WSTYPE=gksqt
```

**Comment by cfelder** [comment]
You can also use either ``GRDIR`` as Josef Heinen already pointed out or ``GKS_FONTPATH`` see C code below:
```C
int gks_open_font(void)
{
  const char *path;
  char fontdb[MAXPATHLEN];
  int fd;

  path = gks_getenv("GKS_FONTPATH");
  if (path == NULL)
    {
      path = gks_getenv("GRDIR");
      if (path == NULL)
        path = GRDIR;
    }
  strcpy(fontdb, (char *) path);
#ifndef _WIN32
  strcat(fontdb, "/fonts/gksfont.dat");
#else
  strcat(fontdb, "\\FONTS\\GKSFONT.DAT");
#endif
  fd = gks_open_file(fontdb, "r");

  return fd;
}
```

Furthermore we are providing linux packages using the open build service.
* [gr](https://software.opensuse.org/download.html?project=science:gr-framework&package=gr)
* [python-gr](https://software.opensuse.org/download.html?project=science:gr-framework&package=python-gr)

**Comment by akiross** [comment]
Sorry for the late answer.

@jheinen I tried with those envvar, but I get no output at all: the program runs with no visible window or message.

@cfelder thanks, I tried using GKS_FONTPATH but I continue to see black boxes in place of text. Thanks also for the other info.

I don't know if this could be relevant, but I am on Wayland.

**Comment by danielkaiser** [comment]
An issue with black text background in the X11 output should be solved with this commit: a9a7337b8c872f889251c5dd4e67d2c887bca0f7

Could you please check if this fix also solves your problem?

**Comment by danielkaiser** [comment]
I'm closing this issue because it should be fixed since the [v0.40.0](https://github.com/sciapp/gr/releases/tag/v0.40.0) release. If you still encounter this problem feel free to reopen this issue.
